filament/backend/src/opengl/OpenGLDriver.cpp

```diff
@@ -681,11 +681,16 @@ void OpenGLDriver::setRasterStateSlow(RasterState rs) noexcept {
     }
 
     // depth test
-    depthFunc(getDepthFunc(rs.depthFunc));
+    if (rs.depthFunc == RasterState::DepthFunc::A && !rs.depthWrite) {
+        disable(GL_DEPTH_TEST);
+    } else {
+        enable(GL_DEPTH_TEST);
```

Comment:
So we can be here if depthFunc is set to ALWAYS and depthWrite is true. Why enable depth testing if depthFunc is ALWAYS?
It makes sense to enable the depthMask but not the depthTest. I think it should be:
```
else {
  if (rs.depthFunc == DepthFunc::A) disable(GL_DEPTH_TEST)
  else enable(GL_DEPTH_TEST)
}
```

Reply:
when GL_DEPTH_TEST is disabled, writes are disabled too!!!